setup.py

```diff
@@ -674,6 +674,19 @@ def make_relative_rpath(path):
                        language='c')
         extensions.append(DL)
 
+        # FIXME: This is a hacky prototype.
```

Comment:
Yeah, I don't think there's any reason to put this in a separate Python extension library.

Reply:
The two nice properties that it provides are:
1) It won't break the build on unsupported platforms.
2) It localizes the extension to the timer folder which makes it super easy to back test on earlier versions of PyTorch.

But feel free to suggest an alternative if you think it belongs somewhere else.

Reply:
Maybe you can use `load_inline` or `load` https://pytorch.org/docs/master/cpp_extension.html?highlight=load_inline#torch.utils.cpp_extension.load_inline, see example use  in the tests https://github.com/pytorch/pytorch/blob/da7863f46bff365499730244a3198a9d4b201f7c/test/test_cpp_extensions_jit.py#L224-L243

Reply:
> It won't break the build on unsupported platforms.

I mean, normally when we have platform specific code we use preprocessor macros to control if it is compiled or not...

> It localizes the extension to the timer folder which makes it super easy to back test on earlier versions of PyTorch.

To me, this is a strong argument that Timer as a whole should live entirely out of tree. If you need to do *post-facto* backtesting it doesn't really make any sense to have critical benchmarking code (which, btw, you want to be the same for all older versions, even if you had fixed bugs midway through) tied to the version you're backtesting.

The alternative is to just have this be yet another module compiled in `_C`, and preprocessored out to raise errors when called on Windows

Reply:
> I mean, normally when we have platform specific code we use preprocessor macros to control if it is compiled or not...

Actually, that is a very reasonable point. For the control bindings we should probably always compile *something*, even if it's just a "this is unsupported" error. 

> To me, this is a strong argument that Timer as a whole should live entirely out of tree. If you need to do post-facto backtesting it doesn't really make any sense to have critical benchmarking code (which, btw, you want to be the same for all older versions, even if you had fixed bugs midway through) tied to the version you're backtesting.

I'd like to see deeper integration between `Timer` and the runtime in the future (and at some point that will mean the copy-paste folder trick won't work), but I see this as a practical compromise in the interim. Having it in tree is also convenient because it doesn't require any other installs. (which I think would harm adoption.)

For the build, I'm going to try @ngimel's suggestion and see how that goes.

Reply:
Thanks to @ngimel for pointing out that if we're JIT-ing the bindings, we can rely on the client `callgrind.h` and `valgrind.h` and we don't have to fork them here. Yay!